chat-core/src/MongoDbDatabaseConnection.ts

```diff
@@ -0,0 +1,43 @@
+import { DatabaseConnection } from "./DatabaseConnection";
+import { Page } from "./Page";
+import { MongoClient, Db } from "mongodb";
+
+export interface MakeMongoDbDatabaseConnectionParams {
+  connectionUri: string;
+  databaseName: string;
+}
+
+/**
+  Constructs connection to MongoDB database.
+ */
+export function makeMongoDbDatabaseConnection({
+  connectionUri,
+  databaseName,
+}: MakeMongoDbDatabaseConnectionParams): DatabaseConnection & {
+  mongoClient: MongoClient;
+  db: Db;
+} {
+  const mongoClient = new MongoClient(connectionUri, {
+    serverSelectionTimeoutMS: 30000,
+  });
+  const db = mongoClient.db(databaseName);
+  return {
+    mongoClient,
+    db,
+    async drop() {
```

Comment:
I still think this should have been called something like `debugDropDatabase()` just to really make clear that this should not be used in non-test code.

Actually, we might want to refactor this out and have the tests instead use MongoDB directly to drop the test databases.

```js
makeTestMongoDbDatabaseConnection(...) {
  const connection = makeMongoDbDatabaseConnection(...);
  return { ...connection, drop() { /* implement drop logic here */ } };
}

Reply:
i think i'll just note in the TSDoc comment `@internal`

Reply:
I'm not 100% satisfied but this isn't something we have to fix today. If someone is calling drop() in production code then...